bundle/config/validate/interpolation_in_auth_config.go

```diff
@@ -0,0 +1,77 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/auth"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/databricks/cli/libs/dyn/dynvar"
+)
+
+type noInterpolationInAuthConfig struct{}
+
+func NoInterpolationInAuthConfig() bundle.Mutator {
+	return &noInterpolationInAuthConfig{}
+}
+
+func (f *noInterpolationInAuthConfig) Name() string {
+	return "validate:interpolation_in_auth_config"
+}
+
+func (f *noInterpolationInAuthConfig) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	authFields := []string{
+		// Generic attributes.
+		"host", "profile", "auth_type", "metadata_service_url",
+
+		// OAuth specific attributes.
+		"client_id",
+
+		// Google specific attributes.
+		"google_service_account",
+
+		// Azure specific attributes.
+		"azure_resource_id", "azure_use_msi", "azure_client_id", "azure_tenant_id",
+		"azure_environment", "azure_login_app_id",
+	}
+
+	diags := diag.Diagnostics{}
+
+	for _, fieldName := range authFields {
+		p := dyn.NewPath(dyn.Key("workspace"), dyn.Key(fieldName))
+		v, err := dyn.GetByPath(b.Config.Value(), p)
+		if dyn.IsNoSuchKeyError(err) {
+			continue
+		}
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		if v.Kind() == dyn.KindInvalid || v.Kind() == dyn.KindNil {
+			continue
+		}
+
+		vv := v.MustString()
```

Comment:
This is a variant of the bug that happened in v0.241.0.

You can do the following for the same effect (and skip the conditional above):
```go
vv, ok := v.AsString()
if !ok {
  continue
}
```

Reply:
I figured it was not a problem here because the list of values is statically defined in the code itself, and thus, it did not run the risk of panicking. 

Changed however in https://github.com/databricks/cli/pull/2412.